src/llama.cpp

```diff
@@ -17448,6 +17495,40 @@ void llama_numa_init(enum ggml_numa_strategy numa) {
     }
 }
 
+void llama_attach_threadpool(
+             struct llama_context * ctx,
+        ggml_compute_threadpool_t   threadpool) {
+    ctx->threadpool = threadpool;
+}
```

Comment:
llama.cpp should create a threadpool automatically if the user does not supply one, otherwise a new one will be created with each call to `ggml_graph_compute`.

Reply:
Ah. We can either do it in llama.cpp or add logic to keep the temporary threadpool around (ie make it non-temporary) in ggml_graph_compute(). Now that you mentioned it, probably makes more sense to do it in llama.cpp and attach to the context. Will do.